llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -2516,6 +2516,11 @@ void VPScalarIVStepsRecipe::print(raw_ostream &O, const Twine &Indent,
 }
 #endif
 
+bool VPWidenGEPRecipe::usesFirstLaneOnly(const VPValue *Op) const {
+  assert(is_contained(operands(), Op) && "Op must be an operand of the recipe");
+  return vputils::isSingleScalar(Op);
+}
+
```

Comment:
Could you add what assertion got triggered? It is not immediately clear to me how this prevents assertions.